### PR TITLE
feat: add a configurable KL penalty to the PPO example

### PR DESCRIPTION
**Pull request by Moviw**
## 背景

PPO 章节给出的目标函数包含 KL penalty，并明确提到示例代码保留了这一项。但 `ppo.py` 原来的损失只包含 clipped policy loss、value loss 和 entropy bonus，没有对应的 KL 项。

## 实现

新增 `kl_coef` 参数，最终最小化的损失为：

    loss = policy_loss
        + value_coef * value_loss
        - entropy_coef * entropy
        + kl_coef * KL(old_policy || current_policy)

  KL 的方向与章节公式一致。CartPole 使用离散动作策略，因此这里通过两个 Categorical 分布，对所有动作计算精确 KL，再对 minibatch 中的状态取平均；不是只根据采样动作的 log probability 估计 KL。

旧策略的 logits 在 `update()` 开始、第一次优化之前保存，并从计算图中分离。在后续所有 minibatch 和 epoch 中都使用同一份快照，避免 KL 的参考策略随更新一起变化。梯度只通过当前策略传播。

## 参数和行为

- `kl_coef` 默认是 `0.0`，保留原调用方式以及不带 KL penalty 的优化目标。
- 可运行示例显式设置 `kl_coef = 0.1`，用于演示这一项；该值没有经过性能调优。
- `update()` 返回的统计信息新增 `kl`，记录各 minibatch 更新前的平均 KL，不代表整个更新结束后的最终 KL。

## 验证

新增 4 项 CPU 回归测试，覆盖：

- KL 结果与 `sum(p_old * (log p_old - log p_new))` 一致，方向为旧策略到新策略。
- 第一次优化前 KL 为零；策略更新后 KL 非零，旧分布在跨 minibatch、跨 epoch 更新时保持不变且不参与梯度计算。
- 非零系数确实影响策略参数更新；系数为零时，将 KL 项置零不改变更新结果。

运行命令：

    MPLBACKEND=Agg OMP_NUM_THREADS=1 python -m unittest discover \
       -s content/docs/02-on-policy-policy-based/ppo \
       -p test_ppo.py -v

在安装了 PyTorch、Gymnasium、NumPy 和 Matplotlib 的环境中，4 项测试通过；`git diff --check` 通过。

### CODE REVIEW

**Comment by lubludrova** [comment]
thanks!
